### PR TITLE
PROF-10073: Add support for datadog.profiling

### PR DESCRIPTION
**Pull request by szegedi**
#### What this PR does / why we need it:

Adds support for `datadog.profiling` variable to enable Continuous Profiler in pods.

Allows Continuous Profiler to be enabled (or disabled) by Helm Charts config on the controller pod by setting an env var on it. The cluster-agent, running on the controller pod, will read that env var and use its value to mutate the configuration of other pods to set the env vars that will activate profiling within the tracer/profiler client libraries.

* This PR depends on cluster-agent interpreting the environment variable: https://github.com/DataDog/datadog-agent/pull/27185, slated for 7.57.0 agent release. Earlier cluster agents will ignore the setting.
* Sister PR of https://github.com/DataDog/datadog-operator/pull/1271
* This PR is modeled after https://github.com/DataDog/helm-charts/pull/1389 which did pretty much the same SSI enablement for ASM products. 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] `CHANGELOG.md` has been updated
- [x] Variables are documented in the `README.md`
- [ ] ~~For Datadog Operator chart or value changes update the test baselines (run: `make update-test-baselines`)~~ _(Not a Datadog Operator change)_

## Testing

Tested running helm from command line, all results as expected:
```
 % helm template ./charts/datadog --set datadog.profiling.enabled=auto | grep PROFILING -A 1
          - name: DD_ADMISSION_CONTROLLER_AUTO_INSTRUMENTATION_PROFILING_ENABLED
            value: "auto"

% helm template ./charts/datadog --set datadog.profiling.enabled=true | grep PROFILING -A 1
          - name: DD_ADMISSION_CONTROLLER_AUTO_INSTRUMENTATION_PROFILING_ENABLED
            value: "true"

% helm template ./charts/datadog --set datadog.profiling.enabled=false | grep PROFILING -A 1
          - name: DD_ADMISSION_CONTROLLER_AUTO_INSTRUMENTATION_PROFILING_ENABLED
            value: "false"

% helm template ./charts/datadog --set datadog.profiling.enabled=null | grep PROFILING -A 1

% helm template ./charts/datadog | grep PROFILING -A 1

% 
```

### CODE REVIEW

File: charts/datadog/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: datadog
-version: 3.69.3
+version: 3.69.4
```
Comment (r1viollet) [inline charts/datadog/Chart.yaml:3]: NIT: is this a minor update ?

**Review by r1viollet — APPROVED** [review]
LGTM, other than a minor question on versions and a lack of knowledge on how these are managed

File: charts/datadog/Chart.yaml (hunk 1 above)
Comment (szegedi) [reply charts/datadog/Chart.yaml:3]: Hopefully someone from container-ecosystems will be able to give us a definitive answer on this.

**Review by khewonc — APPROVED** [review]

**Pull request merged**